File: backend/src/main/java/amos/group3/gitmodconfig_backend/models/ConfigurationRepositoryModel.java
package amos.group3.gitmodconfig_backend.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
public class ConfigurationRepositoryModel {
    private String name;

    private SubmoduleModel[] submodules;

    private boolean auto_init;
}

File: frontend/src/app/models/ConfigurationRepositoryModel.ts
import { Submodule } from "./Submodule";

export class ConfigurationRepositoryModel {
  name: string;
  submodules: Submodule[];

  constructor(name: string, subModules: Submodule[]) {
    this.name = name;
    this.submodules = subModules;
  }
}
